### ISSUE TITLE
stern not working against AWS EKS Cluster

### ISSUE THREAD

**Issue by bobhenkel** [wercker/stern#80]
When I run stern against an EKS cluster(k8s 1.10.2) when I have admin privileges I get this:
```
> stern --kubeconfig ~/.kube/config restic --namespace heptio-ark                                                                                                                                                  
failed to set up watch: failed to set up watch: unknown (get pods)
```

I'm able to use kubectl logs and see pod logs, looks like stern might be using another api  to get the data?

Any help would be appreciated.

**Comment by PaigingAlyssa** [comment]
+1

**Comment by michalschott** [comment]
bump

**Comment by qrevel** [comment]
this is right: `failed to set up watch: failed to set up watch: the server has asked for the client to provide credentials (get pods)`

**Comment by mattmoyer** [comment]
Support for kubeconfig files that use the exec plugin (like the [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator) ones) was added in [client-go v7.0](https://github.com/kubernetes/client-go/commit/a3bffe5d33a89e42927d99f05ba84ce62504a872). It looks like stern is currently using [client-go v5.0.1](https://github.com/wercker/stern/blob/08066e41d7b7c0a41b167021793434275ddb0466/vendor/vendor.json#L1670) so upgrading that should do the trick.

**Comment by markxnelson** [comment]
Hi all, we are not able to merge the PR from @mattmoyer yet, but the change seems fairly minor and he has given us consent to go independently create a fix/PR.  Would you all be able to test it and give feedback?

**Comment by markxnelson** [comment]
I opened a PR #82 to attempt to fix this.  Still a WIP.  Hoping to get feedback from watchers.  Will also fire up a cluster and try to reproduce the issue and test this fix myself.

**Comment by ma-caylent** [comment]
I know it's pretty old, but I'm facing the same issue with EKS 1.14.
